apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,694 @@
+  end.
+
+is_enabled([]) ->
+  false;
+
+is_enabled([{_, enabled}|_]) ->
+  true;
+
+is_enabled([{_, _}|Pools]) ->
+  is_enabled(Pools).
+
+send_check(State) ->
+  Ref = make_ref(),
+  erlang:send_after(?TIMEOUT, self(), {check, Ref}),
+  State#state{check_ref = Ref}.
+
+stop_pool([{Pool, enabled}|_], NodeToId) ->
+  stop_pool(Pool, NodeToId);
+
+stop_pool([_|Pools], NodeToId) ->
+  stop_pool(Pools, NodeToId);
+
+stop_pool(Pool, NodeToId) ->
+  case maps:find(Pool, NodeToId) of
+    error ->
+      ok;
+    {ok, Id} ->
+      aws_cli:stop_instance(Id)
+  end.
+
+start_pool(Pool, NodeToId) ->
+  case maps:find(Pool, NodeToId) of
+    error ->
+      ok;
+    {ok, Id} ->
+      aws_cli:start_instance(Id)
+  end.
+
+-ifdef(TEST).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-define(setup(F), {setup, fun setup_/0, fun cleanup_/1, F}).
+
+setup_() ->
+  lager:start(),
+  ets:new(video, [public, named_table, {keypos, #current_task.id}]).
+
+cleanup_(_) ->
+  ets:delete(video),
+  application:stop(lager).
+
+choose_pool_test_() ->
+  [
+    {"handle state without pools",
+      ?setup(
+        fun choose_pool_no_pool_t_/1
+      )
+    },
+    {"handle state without tasks",
+      ?setup(
+        fun choose_pool_no_tasks_t_/1
+      )
+    },
+    {"handle ordinary state",
+      ?setup(
+        fun choose_pool_t_/1
+      )
+    }
+  ].
+
+choose_pool_no_pool_t_(_) ->
+  T = fun(Id) -> #task{id = Id, category = video, file = #file{}} end,
+  Q = fun(Id) -> fyler_queue:in(T(Id), fyler_queue:new()) end,
+  update_ets(T(1)),
+  update_ets(T(2)),
+  State1 = #state{},
+  State2 = #state{retasks = Q(1)},
+  State3 = #state{tasks = Q(1)},
+  State4 = #state{retasks = Q(1), tasks = Q(2)},
+  [
+    ?_assertEqual({zero, State1}, choose_pool(State1)),
+    ?_assertEqual({tasks, State2}, choose_pool(State2)),
+    ?_assertEqual({tasks, State3}, choose_pool(State3)),
+    ?_assertEqual({tasks, State4}, choose_pool(State4))
+  ].
+
+choose_pool_no_tasks_t_(_) ->
+  State1 = #state{pools = [{a, disabled}]},
+  State2 = #state{pools = [{b, enabled}]},
+  State3 = #state{pools = [{a, disabled}, {b, enabled}]},
+  [
+    ?_assertEqual({zero, State1}, choose_pool(State1)),
+    ?_assertEqual({pools, State2}, choose_pool(State2)),
+    ?_assertMatch({pools, #state{pools = _}}, choose_pool(State3))
+  ].
+
+choose_pool_t_(_) ->
+  T = fun(Id) -> #task{id = Id, category = video, file = #file{}} end,
+  Q = fun(Id) -> fyler_queue:in(T(Id), fyler_queue:new()) end,
+  update_ets(T(1)),
+  update_ets(T(2)),
+  State1 = #state{tasks = Q(1), pools = [{a, disabled}]},
+  State2 = #state{tasks = Q(1), pools = [{b, enabled}]},
+  State3 = #state{tasks = Q(1), pools = [{a, disabled}, {b, enabled}]},
+  State4 = #state{tasks = Q(1), pools = [{a, enabled}, {b, enabled}]},
+  State5 = #state{retasks = Q(2), pools = [{a, disabled}]},
+  State6 = #state{retasks = Q(2), pools = [{b, enabled}]},
+  State7 = #state{retasks = Q(2), tasks = Q(1), pools = [{a, enabled}, {b, enabled}]},
```

Comment:
Line 683 is too long:   State7 = #state{retasks = Q(2), tasks = Q(1), pools = [{a, enabled}, {b, enabled}]},.